ui/src/app/shared/models.ts

```diff
@@ -955,3 +955,13 @@ export interface LinkInfo {
 export interface LinksResponse {
     items: LinkInfo[];
 }
+
+export interface UserMessages {
+    appName: string;
+    msgKey: string;
+    messages?: string;
```

Comment:
This is just more of a question.  It seems like you want to make this general so that you will have more messages to show in the future, but right now you just have the one message ("Since the number of pods....").  Do you see how different messages per msgKey will be used later on?

Note that the message itself is stored in the preferences.  Potentially, if you have X apps that have greater than 15 pods, then you will have X number of entries in the array.

Reply:
I envision showing the messages at several icons under certain conditions, such as when the service has degraded, or an alert advising the user to check the resource status/rollout extension UI/analysis result. This could also be used to broadcast new product features to end users. 

 

Here is my hypothesis regarding performance. 

for an example

If each app displays 10 messages, and the user only accesses x apps (assuming all apps fulfill the condition), then there will be 10x objects stored in array. However, the x should not be a large number for a given user, since most users manage a smaller number of apps through the UI. Since the user prefers to store the data in the browser, the browser's memory footprint should be relatively small.

Let me know what you think. I'm open to modify the code, if there is a better approach. Idea is to show the message once per app for a given user.

Reply:
Not sure if the banner is something that you want to use instead for announcements like that.

Reply:
Banner is one way to provide the information to the user but I am thinking to use this msg property for other use-cases where we can show tooltip to user when specific condition is met like resource is degraded or inform  user to  click on the link  to view failed resources. If you think this approach is too complex I can simply and show the message once to user and not store in the preference

Reply:
That's ok. It's just that the messages will be 'fixed' and if you want to modify or add any new messages, then it will have to be done in an Argo CD release. eg. they won't be configurable, like for the banner messages.

LGTM for what you want to achieve.